js/main.js

```diff
@@ -365,43 +303,113 @@ function updateLegendDataSource(dataSource) {
-    initMarkers(json);
-    initControls();
-    map.addLayer(unclassifiedGroup);
-    map.addLayer(nowGroup);
-    updateLayers();
-    updateDocumentTitle(cityName);
-    updateUrlHash(cityName);
+function setCity(city) {
+    var filename = 'cities/' + city + '.json';
+    $.getJSON(filename, function(json) {
+        positionMap(json.metadata.map_initialization);
+        updateLegendDataSource(json.metadata.data_source);
+        initMarkers(json);
+        initControls();
+        map.addLayer(unclassifiedGroup);
+        map.addLayer(nowGroup);
+        updateLayers();
+        updateUrlHash(city);
+        document.title = 'Wo ist Markt in ' + cityIdToLabel(city) + '?';
+        $('#dropDownCitySelection').val(city).trigger('change', true);
+    }).fail(function() {
+        console.log('Failure loading "' + filename + '".');
+        if (city !== DEFAULT_CITY) {
+            console.log('Loading default city "' + DEFAULT_CITY +
+                        '" instead.');
+            setCity(DEFAULT_CITY);
+        }
+    });
 }
 
+
 /*
- * Forces reloading when URI hash changed.
+ * Load the IDs of the available cities.
+ *
+ * Returns a jQuery `Deferred` object that resolves to an array of city IDs.
  */
+function loadCityIDs() {
+    var d = $.Deferred();
+    $.get(CITY_LIST_API_URL, function(result) {
+        var ids = [];
+        $.each(result, function(_, file) {
+            if (file.name.indexOf('.json', file.name.length - 5) !== -1) {
+                ids.push(file.name.slice(0, -5));
+            }
+        });
+        d.resolve(ids);
+    }).fail(function() {
+        d.fail();
+    });
+    return d;
+}
+
+
 $(window).on('hashchange',function() {
-    window.location.reload(true);
+    setCity(getCityName());
 });
 
+
 $(document).ready(function() {
-    initMap();
-    initLegend();
-    var cityName = getCityName();
-    var marketDataFileName = getMarketDataFilePath(cityName);
-    $.getJSON(marketDataFileName, function(json) {
-        init(json, cityName);
-    }).fail(function() {
-        console.log("Failure loading '" + marketDataFileName + "'. Loading market file for default city (" + DEFAULT_CITY + ") instead.");
-        cityName = DEFAULT_CITY;
-        marketDataFileName = getMarketDataFilePath(cityName);
-        $.getJSON(marketDataFileName, function(json) {
-            init(json, cityName);
-        }).fail(function() {
-           console.log("Failure loading default market file.");
+    var tiles = new L.TileLayer(TILES_URL, {attribution: ATTRIBUTION});
+    map = new L.Map('map').addLayer(tiles);
+    var legend = L.control({position: 'bottomright'});
+    var dropDownCitySelection = $('#dropDownCitySelection');
+    legend.onAdd = function () { return L.DomUtil.get('legend'); };
+
+    // Populate dropdown
+    loadCityIDs().done(function(cityIDs) {
+        for (var i = 0; i < cityIDs.length; i++) {
+            dropDownCitySelection.append(
+                $('<option></option>').val(cityIDs[i])
```

Comment:
Here `cityIDs[i]` should be a local variable. Also the whole statement could be moved into a function such as `createSelectOption(cityId)` to make it very obvious.